### DISCUSSION TITLE
Disable the check the maxParamLength

### DISCUSSION THREAD

**Discussion by saugatadutta** [fastify/fastify#3234] [Q&A] [answered]
### `maxParamLength`
You can set a custom length for parameters in parametric (standard, regex, and multi) routes by using `maxParamLength` option, the default value is 100 characters.<br>
This can be useful especially if you have some regex based route, protecting you against [DoS attacks](https://www.owasp.org/index.php/Regular_expression_Denial_of_Service_-_ReDoS).<br>
*If the maximum length limit is reached, the not found route will be invoked.*

The above check creates an issue while generating prometheus metrics, it create metrics with the `route` as the entire request, raw,url which can be an unusually long string, thereby creating wild card metrics and causing memory issues. Even though application has implementation to replace wildcards with static text in metrics, but due to this check, that code is never invoked and ratify throws the 404: Not found error thereby creating metric like below

`http_request_duration_seconds_count{method="GET",route="/test/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzZXNzaW9uIjp7InN0YXR1cyI6IkF1dGhlbnRpY2F0ZWQiLCJhdXRoZW50aWNhdGlvbkFnZW50IjoiRmFsYWJlbGxhIiwiYXV0aGVudGljYXRpb25TY2hlbWEiOiJQQVNTV09SRCJ9LCJ1c2VyIjp7ImF0Z1VzZXJJZCI6IjExNzAzMDM4NjU2IiwidXNlcklkIjoiZjY1Yzg4ZjAtZDVmNy0xMWViLWFmZDAtN2QwMWExNDg0ZWRlIiwidXNlck5hbWUiOnsiZmlyc3ROYW1lIjoiSmVzw7pzIiwibGFzdE5hbWUxIjoiTWlyYW5kYSJ9LCJwcmltYXJ5UGhvbmUiOnsibnVtYmVyIjoiMzU2NDE5OTUiLCJjb3VudHJ5Q29kZSI6IjU2In0sImFjY291bnQiOnsiaWQiOiJmNjVlNWRiMC1kNWY3LTExZWItYWZkMC03ZDAxYTE0ODRlZGUiLCJ0eXBlIjoiSU5ESVZJRFVBTCJ9LCJkb2N1bWVudCI6eyJpZCI6IjIwNjEwNTQ1LTMiLCJ0eXBlIjoiUlVUIiwiY291bnRyeSI6IkNMIn0sImVtYWlsIjp7ImVtYWlsSWQiOiJhbGV4aXNzLmFtOTc2NEBnbWFpbC5jb20ifSwidXNlclR5cGUiOiJpZGVudGlmaWVkIiwic2VnbWVudHMiOltdfSwiaWRlbnRpdHlWYWxpZGF0b3IiOiJDSUFNIiwiaXNzIjoiQVVUSE4iLCJpYXQiOjE2MjcxMTI4NzF9.",status_code="404"} 2`

### whereas a standard metric looks like this
`http_request_duration_seconds_count{method="GET",route="/test/:id",status_code="404"} 2`

Is there any workaround to handle this ?

**Comment by mcollina** [comment, marked as answer]
Couldn't you increase the maxParamLength to a higher value?

**Reply by saugatadutta** [reply]
@mcollina  we did increase the value and it works, but the problem statement remains, if a request is received where the defined value is breached, then this issue will re-occur.

**Reply by mcollina** [reply]
i'm not sure I understand the question.

**Reply by saugatadutta** [reply]
Let me explain, default value of maxParamLength=100, so when the paramLength increases 100, fastify will invokes the not found route and at this point the metric generated is like the one mentioned above.

Now we have increased the value to 1000, so this check will not be invoked as long as the paramLength is < 1000, But once the paramLength increases 1000, the same check will be invoked to protect against the DoS attack and again the metric generated will be a long route.

So I was wondering if there is any way to disable this check so that this check is never invoked, thereby not generating those wildcard metrics.

**Reply by mcollina** [reply]
You would need to set a value.
I would pass a `16 * 1024` (16KB) - It's significantly high.

**Reply by saugatadutta** [reply]
Thanks, we will use that.
